### PR TITLE
microcom: new recipe

### PR DESCRIPTION
**Pull request by Bastian-Krause**
Pengutronix' microcom comes with more features than busybox' microcom, e.g. "telnet mode" as specified in rfc2217.

### CODE REVIEW

**Comment by Bastian-Krause** [comment]
CI build error should be fixed with #99.

**Comment by Bastian-Krause** [comment]
Rebased on master. Should fix build error fixed by #99.

File: recipes-devtools/microcom/microcom_2019.01.0.bb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+DESCRIPTION = "Minimalistic terminal program for communicating with devices over a serial connection"
+HOMEPAGE = "https://github.com/pengutronix/microcom"
+LICENSE = "GPLv2"
```
Comment (ejoerns) [inline recipes-devtools/microcom/microcom_2019.01.0.bb:3]: Please use SPDX license identifier here (see `scripts/contrib/convert-spdx-licenses.py`)

**Comment by Bastian-Krause** [comment]
Force-pushed:
- license: GPLv2 -> GPL-2.0-or-later (SPDX)
- drop md5sum

File: recipes-devtools/microcom/microcom_2019.01.0.bb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+DEPENDS = "readline"
+
+SRC_URI = "https://github.com/pengutronix/microcom/releases/download/v${PV}/microcom-${PV}.tar.xz"
+SRC_URI[sha256sum] = "87e6bacc545d110ae2bc4443e29d7ccac3c5737a40056d6eef1231e3d8667210"
+
+EXTRA_OECONF = "--enable-largefile"
+
+inherit autotools update-alternatives
+
+PACKAGECONFIG[can] = "--enable-can,--disable-can"
```
Comment (ejoerns) [inline recipes-devtools/microcom/microcom_2019.01.0.bb:14]: One thing I missed during first round: Should this be enabled or disabled by default?

Anyway we should provide a

    PACKAGECONFIG ??= ""

or

    PACKAGECONFIG ??= "can"

Comment (Bastian-Krause) [reply recipes-devtools/microcom/microcom_2019.01.0.bb:15]: I think most people won't need it, so let's keep it disabled by default.

**Comment by Bastian-Krause** [comment]
Force-pushed:
- added empty default PACKAGECONFIG

**Comment by ejoerns** [comment]
Note: backported to 'honister' branch

**Pull request merged**